[FIX] web_editor, website: stop snippet animations before cloning

rdeodoo opened the pull request:
Before this commit, when a snippet was cloned, it was not destroyed.
The flow was:
1. Clone the snippet (DOM) - web_editor trigger `snippet_cloned`
2. Start cloned snippet animation - website listen `snippet_cloned`

This was causing bugs, eg facebook snippet for which the animation is to create
an iframe containing the facebook widget.
When cloning this snippet, its iframe would be cloned to, then its animation
would be started, creating another iframe.

To fix such flows, the snipper cloning should be:
1. Destroy the snippet
2. Clone the snippet (DOM) - web_editor trigger `snippet_cloned`
3. Start cloned snippet animation - website listen `snippet_cloned`
4. Restart the original snippet animation

opw-1951466

rdeodoo commented:
@qsm-odoo Not sure what would be the best way to restart the original snippet that we deactivate before cloning. This does not seems bad

qsm-odoo commented on addons/website/static/src/js/menu/edit.js:
```diff
@@ -158,6 +172,12 @@ var EditPageMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
             editableMode: true,
             $target: ev.data.$target,
         });
+        if (ev.data.$origin) {
```
Not sure why this `if` is useful ?

rdeodoo replied:
As discussed, it is to prevent stable possible custom apps to suddently call this method which will have a new parameter.
If the parameter is not passed, it will reload all #wrapwrap animations.
Safety net for stable, to remove on master as for Odoo standard code we always pass $origin with this PR.

rdeodoo commented:
I retargeted and rebased on 11.0.
The diff needed to be adapted by adding `animation_stop_demand` on `website_root`.
Surprisingly, `animation_stop_demand` trigger up event was only added in 12.0 but some code in 11.0 was calling it, see [here](https://github.com/odoo/odoo/blame/51a9f30e1971155b6315c6bd888d56048191bddd/addons/website/static/src/js/menu/edit.js#L97).
The `_onAnimationStopDemand` method existed but was called only by `ready_to_clean_for_save` trigger up event, not `animation_stop_demand` trigger up event.

Anyway, since facebook snippet was added in 12.0, to test on 11.0 I just added on `registry.slider` `start()` method:
```javascript
this.$test = $("<p>Test</p>");
this.$el.append(this.$test);
```
and in `destroy()` method:
```javascript
if (this.$test) {
    this.$test.remove();
}
```

Without this fix, `<p>Test</p>` will be added twice on cloned snippet while with this fix it will only be added one as it should.

rdeodoo commented on addons/website/static/src/js/content/website_root.js:
```diff
@@ -43,6 +43,7 @@ var WebsiteRoot = BodyManager.extend({
     }),
     custom_events: _.extend({}, BodyManager.prototype.custom_events || {}, {
         animation_start_demand: '_onAnimationStartDemand',
+        animation_stop_demand: '_onAnimationStopDemand',
```
@qsm-odoo Diff needed for 11.0 (only change made since your review for 12.0)

qsm-odoo replied:
Mmh yes, I introduced animation stop on editor save in 12.0, before that all was handled manually in options' cleanForSave methods... Guess it is still fine to merge this in 11.0 anyway :)

rdeodoo commented:
Indeed, done, much clearer

rdeodoo commented:
That was a test to see if you would read again the changes and find the last hidden mispelling.
You succeeded :tada: 
Thanks for help & review

robodoo commented:
Merged, thanks!

rdeodoo commented:
I should strop doing python and do more JS, it seems easier to merge it, no runbot fail :o
